### PR TITLE
Exclude /metrics path from private repos check

### PR DESCRIPTION
**Pull request by galexrt**
Signed-off-by: Alexander Trost <galexrt@googlemail.com>

See issue #68.

### CODE REVIEW

File: handlers.go [hunk 1]
```diff
@@ -132,6 +132,14 @@ func isUserPath(username, path string) bool {
 	return len(path) == len(prefix) || path[len(prefix)] == '/'
 }
 
+func isMetricsPath(path string) bool {
+	prefix := "/metrics"
+	if !strings.HasPrefix(path, prefix) {
+		return false
+	}
+	return len(path) == len(prefix) || path[len(prefix)] == '/'
```
Comment (fd0) [inline handlers.go]: What does this match? `/metrics` and `/metrics/`?

I'd like to standardize on `/metrics` (the literal string, without the slash), so the test is so much simpler: `return path == "/metrics"`. I think we don't need the flexibility here, it just complicates the code.

**Comment by galexrt** [comment]
@fd0 I have updated the PR with your feedback.

File: handlers.go [hunk 2]
```diff
@@ -132,6 +132,14 @@ func isUserPath(username, path string) bool {
 	return len(path) == len(prefix) || path[len(prefix)] == '/'
 }
 
+func isMetricsPath(path string) bool {
+	prefix := "/metrics"
+	if !strings.HasPrefix(path, prefix) {
+		return false
+	}
+	return path == "/metrics"
```
Comment (fd0) [inline handlers.go]: Sorry, this PR fell off my radar. Are the lines 136-139 necessary? I think they can be removed without changing functionality at all. What are your thoughts?

Comment (mholt) [reply handlers.go]: @galexrt I don't really know how the metrics endpoints work, but couldn't this function just be `return strings.HasPrefix(path, "/metrics")`?

Comment (fd0) [reply handlers.go]: The metrics endpoint is just a single path, `/metrics`, so the `==` is appropriate. :0

Comment (galexrt) [reply handlers.go]: I'll change it.

**Pull request merged**